pkg/wal/processor/transformer/wal_transformer_validator.go

```diff
@@ -0,0 +1,100 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package transformer
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/jackc/pgx/v5/pgtype"
+	pglib "github.com/xataio/pgstream/internal/postgres"
+	"github.com/xataio/pgstream/pkg/transformers"
+	"golang.org/x/exp/slices"
+)
+
+func GetTransformerValidationFunc(ctx context.Context, pgurl, snapshoturl string) func(*Transformer) error {
+	// if pgurl or snapshot url is provided, use postgres validator
+	if pgurl == "" {
+		if snapshoturl == "" {
+			return nil
+		}
+		pgurl = snapshoturl
+	}
+	return pgTransformerValidator(ctx, pgurl)
+}
+
+func pgTransformerValidator(ctx context.Context, pgurl string) func(t *Transformer) error {
+	return func(t *Transformer) error {
+		conn, err := pglib.NewConn(ctx, pgurl)
```

Comment:
This is ok since we're only calling this once. If we were performing queries frequently, we'd want to have a struct that would initialise a connection pool that can help reuse resources each time the validation is called. 
I think it's ok to leave it like this for now, but it's something to keep in mind.